cmd/account/mgmt/account-unassign.go

```diff
@@ -373,6 +435,27 @@ func (o *accountUnassignOptions) deleteAttachedPolicies(user string) error {
 	return nil
 }
 
+func (o *accountUnassignOptions) deleteRoles(newAWSClient awsprovider.Client) error {
+
+	listRoleInput := &iam.ListRolesInput{}
+	roles, err := newAWSClient.ListRoles(listRoleInput)
+	if err != nil {
+		return err
+	}
+	// delete all roles except OrganizationAccountAccessRole
+	for _, rolename := range roles.Roles {
+
+		if rolename.RoleName != aws.String("OrganizationAccountAccessRole") {
```

Comment:
This function looks like it's exactly what we are looking for (relating to my previous comment).

However, this specific line I don't think will work the way it's intended, because we're trying to compare pointers.  We'd need to dereference `rolename.RoleName` and compare it to the primitive `"OrganizationAccountAccessRole"` and not what's returned from `aws.String()`.

Reply:
Also, there are other roles that AWS creates that we should NOT delete.

Reply:
my understanding is that since we instantiated a new AWS client with AssumeRole creds, when we perfom ListRoles, it should only list the roles created by whatever account id we provided to the assumeRoleForAccount func

Reply:
Every account (unique account ID) gets a set of roles created by AWS. For example

```
$ aws sts get-caller-identity
{
    "UserId": "AROAU4GZCQTXJOWX2WGWB:tfahlman-sre",
    "Account": "335462696174",
    "Arn": "arn:aws:sts::335462696174:assumed-role/OrganizationAccountAccessRole/tfahlman-sre"
}
```

```
$ aws iam list-roles | jq -r '.Roles[].RoleName'

AccessAnalyzerMonitorServiceRole_SSQKKDWYPI
AWSServiceRoleForAWSLicenseManagerRole
AWSServiceRoleForElastiCache
AWSServiceRoleForElasticLoadBalancing
AWSServiceRoleForGlobalAccelerator
AWSServiceRoleForMarketplaceLicenseManagement
AWSServiceRoleForNetworkFirewall
AWSServiceRoleForOrganizations
AWSServiceRoleForRDS
AWSServiceRoleForSupport
AWSServiceRoleForTrustedAdvisor
BYOCAdminAccess-472x92
BYOCAdminAccess-7f5bbr
BYOCAdminAccess-hwq9l7
BYOCAdminAccess-p7t45f
BYOCAdminAccess-pgt46k
BYOCAdminAccess-r7jr2w
BYOCAdminAccess-sj29s4
BYOCAdminAccess-x7m5gf
customperms-ControlPlane-Role
customperms-Installer-Role
customperms-openshift-cloud-credential-operator-cloud-credential
customperms-openshift-cluster-csi-drivers-ebs-cloud-credentials
customperms-openshift-image-registry-installer-cloud-credentials
customperms-openshift-ingress-operator-cloud-credentials
customperms-openshift-machine-api-aws-cloud-credentials
customperms-Support-Role
customperms-Worker-Role
ManagedOpenShift-ControlPlane-Role
ManagedOpenShift-IAM-Role
ManagedOpenShift-Installer-Role
ManagedOpenShift-openshift-cloud-credential-operator-cloud-crede
ManagedOpenShift-openshift-cluster-csi-drivers-ebs-cloud-credent
ManagedOpenShift-openshift-image-registry-installer-cloud-creden
ManagedOpenShift-openshift-ingress-operator-cloud-credentials
ManagedOpenShift-openshift-machine-api-aws-cloud-credentials
ManagedOpenShift-Support-Role
ManagedOpenShift-Worker-Role
OrganizationAccountAccessRole
rds-monitoring-role
```
That includes both roles that have been user created (by me, and the rosa cli) as well as created by AWS by default when the account was first created. Osdctl should ONLY delete the user created roles